java/connector-framework-internal/src/main/java/org/identityconnectors/framework/impl/api/local/ConnectorPoolManager.java

```diff
@@ -213,51 +214,74 @@ public void shutdown() {
                     new ConnectorPoolKey(impl.getConnectorInfo().getConnectorKey(), impl
                             .getConfigurationProperties(), impl.getConnectorPoolConfiguration());
 
-            // get the pool associated..
-            ObjectPool<PoolableConnector> pool = POOLS.get(key);
-            // create a new pool if it doesn't exist..
-            if (pool == null) {
-                LOG.info("Creating new pool: {0}", impl.getConnectorInfo().getConnectorKey());
-                // this instance is strictly used for the pool..
-                pool = new ObjectPool<>(
-                        new ConnectorPoolHandler(impl, localInfo), impl.getConnectorPoolConfiguration());
-                // add back to the map of POOLS..
-
-                ObjectPool<PoolableConnector> previousPool = POOLS.putIfAbsent(key, pool);
-                // Use the pool made by other thread
-                if (previousPool != null) {
-                    pool = previousPool;
-                }
+            synchronized (POOLS) {

```

Comment:
@semancik I have a question: is it really needed to synchronize on a `ConcurrentMap` as `POOLS`?

Reply:
Not sure about this myself. The code seems to be overly complicated: combining locks, concurrent collections and synchronized keywords. Therefore it is quite hard to check if it is correct. Would be nice to rewrite the code one day. But for now we have to live with what we have inherited.
Which means it is possible that this specific "synchronized" is redundant. Thanks for pointing that out. Additional pair of eyes is always helpful. I can do more experiments with this it next few days. We'll see.

Reply:
Fine, then :+1:

Reply:
In case you come up with further modifications, I don't think it is needed to open a new JIRA - just refer again to BASE-76, thanks.

Reply:
It looks like this particular "synchronized" is not needed. My parallel tests are passing and it seems to be stable. I'll commit a fix.